app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileServiceInstanceRow', {
+    controller: [
+      '$filter',
+      '$routeParams',
+      '$scope',
+      'APIService',
+      'AuthorizationService',
+      'BindingService',
+      'DataService',
+      'ListRowUtils',
+      MobileServiceInstanceRow
+    ],
+    controllerAs: 'row',
+    bindings: {
+        apiObject: '<',
+        serviceClass: '<',
+        state: '<'
+    },
+    templateUrl: 'views/_mobile-service-instance-row.html'
+  });
+
+  function MobileServiceInstanceRow($filter,
+                              $routeParams,
+                              $scope,
+                              APIService,
+                              AuthorizationService,
+                              BindingService,
+                              DataService,
+                              ListRowUtils) {
+    var row = this;
+    var serviceInstanceDisplayName = $filter('serviceInstanceDisplayName');
+    row.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    row.state = row.state || {};
+    var annotationSpace = "org.aerogear.binding." + this.apiObject.metadata.name;
+    var extendedAnnotationSpace = "org.aerogear.binding-ext." + this.apiObject.metadata.name;
+    var watches = [];
+    var bindingsWatch = false;
+    row.bindingsLimit = _.get(row.serviceClass, "spec.externalMetadata.bindingsLimit", 1);
```

Comment:
`row.$onInit`?  Not sure if there is a risk on a race condition here, for things like `row.bindingsLimit`.

Reply:
Possibly applies to anything that deals with `this.apiObject` as well.  

Even if not a race condition issue, its probably a better convention.

Reply:
yes, makes sense. Will update.